Update Docker configuration to latest Ubuntu LTS and dependencies

shivnshshrma opened the pull request:
# Pull Request Description

## Summary
This pull request updates the Docker configuration to use Ubuntu 25.04 LTS as the base image and upgrades all dependencies to their latest stable versions. Additionally, minor typos in the `README.md` file have been corrected for improved clarity.

## Why the Change?
The previous base image, Ubuntu 20.04 LTS, is nearing its end-of-life in April 2025. Upgrading to Ubuntu 25.04 LTS ensures:
- Continued security updates and support.
- Access to the latest features and improvements in the Ubuntu ecosystem.
- Compatibility with updated versions of dependencies.

## Key Changes
1. **Dockerfile:**
   - Updated the base image from `ubuntu:20.04` to `ubuntu:25.04`.
   - Verified and updated all dependencies to their latest stable versions.
   - Ensured compatibility with the new Ubuntu version by testing the build process.

2. **README.md:**
   - Fixed typos and clarified instructions.
   - Updated references to the Ubuntu version where applicable.

## Testing
- Successfully built and tested the Docker image with Ubuntu 25.04.
- Verified that all dependencies work correctly with the updated configuration.
- Tested the SOS application and benchmarks in the updated container to ensure functionality.

## Impact
This update ensures that the project remains secure, up-to-date, and maintainable by aligning with the latest LTS release of Ubuntu.

## Next Steps
- Review and merge this pull request.
- Update any documentation or CI/CD pipelines that reference the old Ubuntu version.

markbrown314 commented:
Thanks @shivnshshrma. Linking with issue #1186. Will review shortly.

markbrown314 commented on scripts/docker/Dockerfile:
```diff
@@ -1,5 +1,4 @@
-FROM ubuntu:20.04
-
+FROM ubuntu:25.04
```
@shivnshshrma we want to target the latest Ubuntu LTS which is currently 24.04.

markbrown314 commented on scripts/docker/Dockerfile:
```diff
@@ -80,7 +77,7 @@ RUN make install
 # Build Portals 4
 WORKDIR $SOS_INSTALL
 RUN git clone --depth 10 https://github.com/regrant/portals4.git portals4
```
Portals 4 is now at https://github.com/sandialabs/portals4
regrant/portals4 aliases sandialabs/portals4, but we should update this to the proper location.

markbrown314 reviewed (changes requested):
Commit message is too long and has an extra space. It should look something like this:
```
Updated Docker configuration
* Uses Ubuntu 24.04 LTS
* Fixed typos in README.md

Issue #1186

Signed-off-by: USE FULL NAME HERE <shivnshshrma@gmail.com>
```

shivnshshrma commented:
Updated docker configuration with all the necessary changes

- Uses Ubuntu 24.04 LTS
- Updated Portals 4 repository URL to the correct location (sandialabs/portals4)

davidozog commented on scripts/docker/Dockerfile:
```diff
@@ -79,8 +78,8 @@ RUN make install
 WORKDIR $SOS_INSTALL
-RUN git clone --depth 10 https://github.com/regrant/portals4.git portals4
-WORKDIR portals4
+RUN git clone --depth 10 https://github.com/sandialabs/portals4.git portals4
```
This latest commit doesn't work for me - it seems related to the temporary fix in #1203 here:
https://github.com/Sandia-OpenSHMEM/SOS/pull/1203/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR638

Should we make that change to portals4 upstream @markbrown314?

Or we could simply clone portals4 at commit https://github.com/sandialabs/portals4/commit/b9083c85d09c8810b726aef2e3cd19355afb5533

Or simply comment this out?  We're not even using it below...

davidozog replied:
Otherwise, thanks for the patch/update @shivnshshrma !

markbrown314 commented:
@shivnshshrma are you still interested in completing this pull request?

shivnshshrma commented:
@markbrown314 yes definitely
So we have to change the ubuntu 24 to ubuntu 22 right?
